sphinx/writers/html.py

```diff
@@ -129,8 +129,10 @@ def depart_desc(self, node: Element) -> None:
     def visit_desc_signature(self, node: Element) -> None:
         # the id is set automatically
         self.body.append(self.starttag(node, 'dt'))
+        self.optional_param_level += 1
```

Comment:
@jakobandersen Could you tell me your opinion on this change, please? Now I'm going to escape the whole of  `desc_signature` node. As a result, consecutive spaces are rendered as-is in HTML browsers. Additionally, non-spaces becomes not foldable. Is there any side-effect to C and C++ domains?